src/RequestException.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Amp\Http\Client;
+
+abstract class RequestException extends HttpException
```

Comment:
Seems like this extra exception is useless?

Reply:
Sort of… it acts as shared code for `InvalidRequestException` and `UnprocessedRequestException`. Certainly open to different suggestions.

Reply:
Where is `::getRequest()` used? Or is it only intended for consumers of the library?

Reply:
I originally intended to use it in `RetryRequests` when an `UnprocessedRequestException` was thrown. A consumer *might* use it to see if the request was modified by interceptors, but I'm not sure if it's necessary.